src/providers/drafts.ts

```diff
@@ -0,0 +1,123 @@
+import { Language, License, VideoCategory } from '@/api/queries'
+import { CommonStore, createStore } from '@/store'
+import { createId } from '@/utils/createId'
+import { readFromLocalStorage } from '@/utils/localStorage'
+
+export type CommonDraftProps = {
+  id: string
+  channelId: string
+  updatedAt: string
+  seen?: boolean
+}
+
+export type Draft = VideoDraft
+
+export type DraftType = 'video'
+
+export type VideoDraft = {
+  type: 'video'
+  title?: string | null
+  description?: string | null
+  isPublic?: boolean | null
+  publishedBeforeJoystream?: string | null
+  hasMarketing?: boolean | null
+  isExplicit?: boolean | null
+  language?: Language['iso'] | null
+  category?: VideoCategory['id'] | null
+  licenseCode?: License['code'] | null
+  licenseCustomText?: License['customText'] | null
+  licenseAttribution?: License['attribution'] | null
+} & CommonDraftProps
+
+export type RawDraft = Omit<Draft, 'id' | 'updatedAt'>
+
+type DraftStoreStateV1 = {
+  allDrafts: Draft[]
+}
+type DraftStoreActions = {
+  addDraft: (draft: RawDraft, explicitId?: string) => Draft
+  updateDraft: (draftId: string, draftProps: RawDraft) => void
+  removeDrafts: (draftIds: string[]) => void
+  removeAllDrafts: (channelId: string) => void
+  markAllDraftsAsSeenForChannel: (channelId: string) => void
+}
+
+export const useDraftStore = createStore<DraftStoreStateV1, DraftStoreActions>(
+  {
+    state: {
+      allDrafts: [], // includes drafts for different channels
```

Comment:
Let's just use `drafts` here, we do the same in uploads, we don't call it `allUploads`